src/plugins/home/server/services/sample_data/lib/sample_dataset_registry_types.ts

```diff
@@ -27,7 +28,11 @@ export enum EmbeddableTypes {
   SEARCH_EMBEDDABLE_TYPE = 'search',
   VISUALIZE_EMBEDDABLE_TYPE = 'visualization',
 }
-export type SampleDatasetProvider = () => SampleDatasetSchema;
+
+export interface SampleDatasetProviderContext {
+  staticAssets: IStaticAssets;
+}
+export type SampleDatasetProvider = (context: SampleDatasetProviderContext) => SampleDatasetSchema;
```

Comment:
Adapted `SampleDatasetProviderContext` to add the `staticAssets` subservice to it, so that providers can use it to generated CDN-friendly hrefs for their assets